BEP7.md

```diff
@@ -0,0 +1,320 @@
+      - [5.2.1 Issue collection](#521-issue-collection)
+      - [5.2.2 Increase total supply](#522-increase-total-supply)
+  - [6. Token Specification](#6-token-specification)
+    - [6.1 Token Properties](#61-token-properties)
+    - [6.2 Token Management Operation](#62-token-management-operation)
+      - [6.2.1 Issue token](#621-issue-token)
+      - [6.2.2 Transfer Tokens](#622-transfer-tokens)
+      - [6.2.3 Freeze Tokens](#623-freeze-tokens)
+      - [6.2.4 Unfreeze Tokens](#624-unfreeze-tokens)
+  - [7. Token Use Case Examples](#7-token-use-case-examples)
+  - [8. License](#8-license)
+  - [9. Reference](#9-reference)
+
+## 1. Summary
+
+This BEP describes a proposal for non-fungible tokens(NFTs) management on the Binance Chain.
+
+## 2. Abstract
+
+BEP-7 Proposal describes a common set of rules for non-fungible token within the Binance Chain ecosystem. It introduces the following details of a non-fungible token on Binance Chain:
+
+- What information makes a non-fungible token on Binance Chain
+- What actions can be performed on a non-fungible token on Binance Chain
+
+### Couple things to keep in mind
+
+- This BEP won't contain "how to implement" in terms of implmentation on the Binance Chain codebase, e.g. which module NFT should be in (this kind of discussion is happening on [cosmos-sdk/issues/4046](https://github.com/cosmos/cosmos-sdk/issues/4046)), because that should be happening in the PR(against to Binance Chain, when it's fully open source, so that dev can send out PR.
+
+- Apparently Binance chain doens't have solidity-such smart contract implemented, so don't expect to see many smart contract implmenations like what you can see in [EIP-721](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md) in this BEP.
+
+## 3. Status
+
+This BEP is under drafting.
+
+## 4. Motivation
+
+Design and issue non-fungible asset on the Binance Chain, as one of the basic economic foundations of the blockchain.
+
+## 5. Collection Specification
+
+### 5.1 Collection Properties
+
+- Owner: the owner adress of the collection.
+
+- Token Name: Token Name represents the long name of the token - e.g. "BinanceAnnualEvent2019"
+
+- Denom: Denom is the identifier(symbol) of the newly issued collection.
+
+- Total Supply: Total supply is the total number of issued NFT tokens belong to this collection.
+
+- Mintable: A boolean value to indicate whether the owner can increase the totalSupply.
+
+- NFTs: An array contains items in NFT interface(described below).
+
+```go
+// Collection of non fungible tokens
+type Collection struct {
+  Owner         string  `json:"owner"`                    // owner of the collection;
+  Name          string  `json:"name"`                     // name of the collection;
+  Denom         string  `json:"denom,string,omitempty"`   // the identifier(symbol) of the newly issued collection;
+  TotalSupply   int64   `json:"totalSupply"`              // total supply of the NFT in this collection;
```

Comment:
Is `TotalSupply` ever going to be different from the length of the array of `NFTs`?

Reply:
@amazingandyyy is there a particular rationale for TotalSupply in64? I know that ERC 1155 gives you the option of minting 1 as a unique NFT or an array for limited "collections". I think that over-specification, in the spec could be confusing as the ability to mint on chain would be meant to allow custom specific use cases whereby an NFT would not nessarily need to declare a total supply unless it was required by an open sea type wallet. Now we could include the specs to give that capability as standard specific to BEP7 that would allow for others to return to our spec as the most cross chain combatible as Loom Network and Wink share cross chain capabilities with Smart Contract side chains that are Ethereum based, or EOS, or TRON.

Reply:
@casmirconsensys I am not sure what do you mean **over-specification** , `int64` simply means it can be any number.

Reply:
@okwme  yes, for example, totalySupply is 1M, and the collection starts from zero token in it, then owner can mint new NFT into the []NFTs away until it hits 1M cap.

Reply:
> @casmirconsensys I am not sure what do you mean **over-specification** , `int64` simply means it can be any number.

This was a question @okwme raised during the call from last week so we wanted to get clarification on what that meant thanks for clearing that up! @amazingandyyy

Reply:
I see, yea, I hope my explanation clarify it.